.github/workflows/build.yaml

```diff
@@ -35,72 +36,23 @@ jobs:
 
     steps:
       - name: Code checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
-      - name: Install Rust toolchain (${{ matrix.rust }})
-        uses: actions-rs/toolchain@v1
+      - name: Install Rust toolchain (stable)
+        uses: dtolnay/rust-toolchain@stable
         with:
-          toolchain: ${{ matrix.rust }}
           target: ${{ matrix.platform.target }}
-          override: true
-          components: clippy
 
-      - name: Build (all crates)
-        uses: actions-rs/cargo@v1
-        with:
-          use-cross: true
-          command: build
-          args: --locked --target=${{ matrix.platform.target }} --workspace --all-targets ${{ matrix.platform.args }}
-
-      - name: Check build did not modify any files
-        run: test -z "$(git status --porcelain)"
-
-  release:
-    name: Build release
-    runs-on: ubuntu-latest
-    strategy:
-      fail-fast: false
-      matrix:
-        rust:
-          - stable
-
-        platform:
-          - target: x86_64-unknown-linux-gnu
-            args: "--features full"
-
-          # For `musl` builds openssl must be vendored
-          - target: x86_64-unknown-linux-musl
-            args: "--features full --features openssl-vendored"
+      - name: Install cross
+        run: cargo install cross --git https://github.com/cross-rs/cross
```

Comment:
We could install cross binaries to reduce build time - threre are tarballs [here](https://github.com/cross-rs/cross/releases/tag/v0.2.5)